app/models/nic/bond.rb

```diff
@@ -44,6 +45,17 @@ def ensure_virtual
     def enc_attributes
       @enc_attributes ||= (super + %w(mode attached_devices bond_options))
     end
+
+    def ensure_mac
+      mac_addresses = attached_devices_identifiers.collect { |a|
+        self.host.interfaces.where(identifier: a).first.mac
+      }
+      self.mac = nil unless mac_addresses.include? self.mac
```

Comment:
I'd combine L53-L56 like this
```ruby
mac_addresses = mac_addresses.compact # actually this should get to previous mac_addresses SQL...
self.mac =  mac_addresses.first unless mac_addresses.include?(self.mac)
```

overall I'm not sure about this, we'd silently override user custom input if it's another mac, I'd rather override the mac only if it's nil, what do you think?